### PR TITLE
Add Pull Dog

### PR DESCRIPTION
**Pull request by ffMathy**
# Quality Standards

Note that we can help you achieve those standards, just try your best and be brave.
We'll guide you to the best of our abilities.

To be on the list, it would be **nice** if entries adhere to these quality standards:

- It should take less than 20 sec to find what is the project, how to install it and how to use it.
- Generally useful to the community.
- A project on GitHub with a well documented `README.md` file and plenty of examples is considered high quality.
- Clearly stating if an entry is related to (Linux) containers and not to Docker. There is an [awesome list](https://github.com/Friz-zy/awesome-linux-containers) for that.
- Clearly stating "what is it" i.e. which category it belongs to.
- Clearly stating "what is it for" i.e. mention a real problem it solves (even a small one). Make it clear for the next person.
- If it is a **WIP** (work in progress, not safe for production), please mention it. (Remember the time before Docker 1.0 ? ;-) )
- Always put the link to the GitHub project instead of the website!

To be on the list, the project **must** have:

- How to setup/install the project
- How to use the project (examples)

If your PR is not merged, we will tell you why so that you may be able to improve it.
But usually, we are pretty relaxed people, so just come and say hi, we'll figure something out together.

# Rules for Pull Request

- Each item should be limited to one link, no duplicates, no redirection (careful with `http` vs `https`!)
- The link should be the name of the package or project or website
- Description should be clear and concise (read it out loud to be sure)
- Description should follow the link, on the same line
- Entries are listed alphabetically, please respect the order
- If you want to add more than one link, please don't do all PR on the exact same line, it usually results in conflicts and your PR cannot be automatically merged...

Please contribute links to packages/projects you have used or are familiar with. This will help ensure high-quality entries.

### CODE REVIEW

**Comment by ffMathy** [comment]
How do I solve the PR?

And I didn't see the message about the tweet, because I clicked "correct README.md" directly in the GitHub UI. Can I still change the message before it goes out?

File: README.md [hunk 1]
```diff
@@ -169,7 +169,7 @@ _Source:_ [What is Docker](https://www.docker.com/why-docker)
 - [Zodiac](https://github.com/CenturyLinkLabs/zodiac) :skull: - A lightweight tool for easy deployment and rollback of dockerized applications. By [@CenturyLinkLabs][centurylinklabs]
 
 ### Deployment and Infrastructure
-
+- [Pull Dog](https://github.com/apps/pull-dog) - A GitHub app that automatically creates Docker-based test environments for your pull requests, from your docker-compose files.
```
Comment (SuperSandro2000) [inline README.md:172]:
```suggestion

- [Pull Dog](https://github.com/apps/pull-dog) - A GitHub app that automatically creates Docker-based test environments for your pull requests, from your docker-compose files.
```

Comment (ffMathy) [reply README.md:173]: Thank you, I fixed this. Sorry for missing it the first time!

**Comment by ffMathy** [comment]
Hmmm, it still fails.

File: README.md (hunk 1 above)
Comment (SuperSandro2000) [reply README.md:173]: no probemo :)

**Comment by SuperSandro2000** [comment]
> Hmmm, it still fails.

It thinks the link to pull dog is a github repo and ``apps`` is the username. The test script probably needs some fixing for this.

**Comment by ffMathy** [comment]
Hmmm. Perhaps "apps" should just be whitelisted altogether in the script - I don't think it's a valid repo name anyway.

What to do next? Can I help fix this?

**Comment by ffMathy** [comment]
The source code is not open, but it is a GitHub App that creates a test environment for each pull request, to try out a feature in the cloud before merging it in.

It does so by using the project's `docker-compose.yml` file - the server is spawned up using a Docker Compose file.

I wrote more about it here: https://medium.com/@mathiaslykkegaardlorenzen/pull-dog-docker-test-environments-for-pull-requests-21af71784bcc

**Comment by ffMathy** [comment]
Also, what is the problem with pointing to apps? Does it really have to be a repository?

File: README.md [hunk 2]
```diff
@@ -170,6 +170,7 @@ _Source:_ [What is Docker](https://www.docker.com/why-docker)
 
 ### Deployment and Infrastructure
 
+- [Pull Dog](https://github.com/apps/pull-dog) - A GitHub app that automatically creates Docker-based test environments for your pull requests, from your docker-compose files.
```
Comment (veggiemonk) [inline README.md:173]: * All entries must be ordered alphabetically (from A to Z)
* It should show in the description that it is not open source!
* This is a better section for this entry: https://github.com/veggiemonk/awesome-docker#testing

Comment (ffMathy) [reply README.md]: I fixed these points :+1: Thank you.

**Pull request merged**